### PR TITLE
[wip] First draft of video infrastructure

### PR DESCRIPTION
**Pull request by jamesianberry**
Starting position, current parameters will need to be adjusted to new AWS environment

### CODE REVIEW

**Comment by jamesianberry** [comment]
Best not to merge until the following has been done:
1. AWS credentials for automation have been set up and secrets added to GitHub
2. Pulumi signed up for, access token created and secret added to GitHub, GitHub App installed
3. Confirm OpenViduSecret secret already in GitHub
4. New Elastic IP in AWS for use by Pulumi infrastructure until ready for production use
5. Change parameters in video\Program.cs to align with email, domain, IP, instance type and keypair

Even so this is still a rudimentary single server build to match current production. Future work to overcome CloudFormation issues and introduce auto-scaling and separate webapp hosting. Will look at building OpenVidu Pro infrastructure to support more sessions although that will be an extra licensing cost.

File: .github/workflows/video-preview.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+name: Video
+on:
+  - pull_request
+jobs:
+  preview:
+    name: Preview
```
Comment (vadi2) [inline .github/workflows/video-preview.yml:6]: Could you make this name be more meaningful? It'll show up on the left side in https://github.com/grahamegrieve/ClinicArrivals/actions.

File: .github/workflows/video-update.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+name: Video
+on:
+  push:
+    branches:
+      - master
+jobs:
+  up:
+    name: Update
```
Comment (vadi2) [inline .github/workflows/video-update.yml:8]: Same here. Perhaps this should also be fixed to only run in this repo and not forks?

File: video/CF-OpenVidu-latest.yaml [hunk 3]
```diff
@@ -0,0 +1,650 @@
...
+  WebsiteURLLE:
+    Description: Use this URL to connect to OpenVidu
+    Value: !Join
+      - ''
+      - - 'https://'
+        - !Ref MyDomainName
+        - ':4443'
+    Condition: NoSelfSigned
+  DemosURL:
+    Description: If you choose to deploy the demos you can access througt this
```
Comment (vadi2) [inline video/CF-OpenVidu-latest.yaml:636]:
```suggestion
    Description: If you choose to deploy the demos you can access them through this
```

File: video/CF-OpenVidu-latest.yaml [hunk 4]
```diff
@@ -0,0 +1,650 @@
...
+    Value: !Join
+      - ''
+      - - 'https://'
+        - !GetAtt
+          - OpenviduServer
+          - PublicDnsName
+  DemosURLLE:
+    Description: If you choose to deploy the demos with your own DNS name you can access througt this
```
Comment (vadi2) [inline video/CF-OpenVidu-latest.yaml:644]:
```suggestion
    Description: If you choose to deploy the demos with your own DNS name you can access through this
```

File: video/Program.cs [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+{
+    static Task<int> Main()
+    {
+        return Deployment.RunAsync( () => {
+
+            // Use a copy of the OpenVidu project's CloudFormation template
+            var template = System.IO.File.ReadAllText("CF-OpenVidu-latest.yaml");
+
+            // Retrieve OpenViduSecret environment variable
+            string OpenViduSecret = Environment.GetEnvironmentVariable("OpenViduSecret");
```
Comment (vadi2) [inline video/Program.cs:19]: Need error checking here if it hasn't been defined

**Review by vadi2 — COMMENTED** [review]
It looks good at an overview!

**Pull request closed without merging**